code/deserializeJSON.m

```diff
@@ -0,0 +1,46 @@
+function data = deserializeJSON(filename)
+%%
+% Reads .JSON files from RC+S and loads data into Matlab. In cases where
+% end closing brackets is missing, will add in order to make file readable.
+% Requires turtle_json toolbox
+%%
+
+% Add turtle_json toolbox to path
+pathToCode = mfilename('fullpath');
+filePath = fileparts(pathToCode);
+addpath(genpath(fullfile(filePath,'toolboxes', 'turtle_json','src')))
+
+% Try loading json file - may need to have a closing bracket added
+start = tic;
+try
+    data = json.load(filename);
+    fprintf('File loaded in %.2f seconds\n',toc(start));
+catch
+    warning('Not able to open file - attmepting fix');
+    fprintf('Defective file %s\n',filename);
+    
+    % Try to fix the file; separate fixes for Adaptive vs all other JSON
+    % file types
+    dat = fileread(filename);
+    [~,jsonFileName,~] = fileparts(filename);
+    
+    if strcmp(jsonFileName,'AdaptiveLog')
+        adaptiveFile = filename;
+        data = jsondecode(fixMalformedJson(fileread(adaptiveFile),'AdaptiveLog'));
```

Comment:
Why is this json loading method different from lines 16 and 36?

Reply:
@roeegilron Can you comment if the different method for adaptive files is still needed?

Reply:
Sorry only getting to this now, but it is needed bcs these files often don't conform to the .json file structure. In addition, TD data files are too big so this bit of code often fails and I have to use a different work around. A better solution would be to just use textscan and regexp in Matlab and fix it without this bit of code.